src/embedding/kd.ts

```diff
@@ -0,0 +1,414 @@
+         */
+        ListeningForDrillableItems = 'listeningForDrillableItems',
+
+        /**
+         * Type represent that the embedded content starts loading.
+         */
+        LoadingStarted = 'loadingStarted',
+
+        /**
+         * Type represent that The user does not have permissions to view or edit the content.
+         */
+        NoPermissions = 'noPermissions',
+
+        /**
+         * Type represent that an operation increasing the height of the hosting iframe is performed.
+         */
+        Resized = 'resized',
+
+        /**
+         * Type represent that A dashboard has been created and saved.
+         */
+        DashboardCreated = 'dashboardCreated',
+
+        /**
+         * Type represent that the content is fully loaded,
+         * and the user has permissions to access the dashboard.
+         */
+        DashboardLoaded = 'loaded',
+
+        /**
+         * The existing dashboard has been updated.
+         */
+        DashboardUpdated = 'dashboardUpdated',
+
+        /**
+         * Type represent that dashboard is saved.
+         *
+         */
+        DashboardSaved = 'dashboardSaved',
+
+        /**
+         * Type represent that dashboard is deleted.
+         *
+         */
+        DashboardDeleted = 'dashboardDeleted',
+
+        /**
+         * The user cancels the creation of the dashboard.
+         */
+        DashboardCreationCanceled = 'dashboardCreationCanceled',
+
+        /**
+         * Type represent that the dashboard is switched to edit mode.
+         */
+        SwitchedToEdit = 'switchedToEdit',
+
+        /**
+         * Type represent that the dashboard is switched to view mode.
+         */
+        SwitchedToView = 'switchedToView',
+
+        /**
+         * Type represent that the platform is down.
+         */
+        Platform = 'platform'
+    }
+
+    /**
+     * List of available commands. This is included in each event sent by KD.
+     */
+    export interface IKdAvailableCommands {
+        /**
+         * Array of avaiable commands types.
+         */
+        availableCommands: GdcKdCommandType[];
+    }
+
+    /**
+     * Save command body sent by outer application
+     */
+    export interface IKdSaveCommandBody {
+        /**
+         * Dashboard title - use as title of new dashboard or rename of saved dashboard
+         */
+        title: string;
+    }
+
+    /**
+     * Saves current dashboard.
+     *
+     * Contract:
+     *
+     * -  if currently edited dashboard IS NOT eligible for save (empty, in-error), then CommandFailed event
+     *    will be posted
+     * -  if the specified title is invalid / does not match title validation rules, then CommandFailed event
+     *    will be posted
+     * -  otherwise dashboard WILL be saved with the title as specified in the body and the DashboardSaved event
+     *    will be posted
+     * -  the DashboardSaved event will be posted even when saving dashboard that has not changed but would
```

Comment:
This contract is not clear. Did you mean that `DashboardSaved` event is always posted event succeed or failed?

Reply:
Only for succeed, We still posted event when content or title of dashboard has not changed